controllers/admin/AdminShopController.php

```diff
@@ -217,14 +217,30 @@ public function renderList()
     public function displayAjaxGetCategoriesFromRootCategory()
     {
         if (Tools::isSubmit('id_category')) {
-            $selected_cat = [(int) Tools::getValue('id_category')];
-            $children = Category::getChildren((int) Tools::getValue('id_category'), $this->context->language->id);
-            foreach ($children as $child) {
-                $selected_cat[] = $child['id_category'];
-            }
+            // Array responsible of selected categories storage.
+            // To make this array construction easier, we use [ (int) 'category_id' => (int) 'category_id'] format.
+            $selected_categories = [];
+
+            // This recursive anonymous function is in charge of building categories tree.
+            $add_children_categories = function (int $category_id) use (&$selected_categories, &$add_children_categories): void {
+                $children = Category::getChildren($category_id, $this->context->language->id);
```

Comment:
When I this behavior I tell myself... mmm maybe we can put this code in a dedicated class and unit test it? 😄

Reply:
:thinking: Interesting idea.
Should I create some `PrestaShopBundle\Utils\Tree` class ? Maybe some class exists already ?

Reply:
Yes for a `Tree` class 👍 it works for me